bin/p2-rm/main.go

```diff
@@ -24,183 +20,90 @@ import (
 	}
 
 	if podIsManagedByRC && *deallocation {
 		err = rm.decrementDesiredCount(rcID)
 		if err != nil {
-			fmt.Fprintf(os.Stderr,
-				"Encountered error deallocating from the RC %s. You may attempt this command again or use `p2-rctl` to cleanup manually.\n%v",
+			return fmt.Errorf("Encountered error deallocating from the RC %s. You may attempt this command again or use `p2-rctl` to cleanup manually.\n%v",
 				rcID,
 				err)
 		}
 	}
 
 	fmt.Printf("%s: successfully removed %s\n", rm.NodeName, rm.PodName)
-}
-
-func sessionName(rcID fields.ID) string {
-	currentUser, err := user.Current()
-	username := "unknown"
-	if err == nil {
-		username = currentUser.Username
-	}
-
-	return fmt.Sprintf("p2-rm:user:%s:rcID:%s", username, rcID)
-}
-
-func (rm *P2RM) decrementDesiredCount(id fields.ID) error {
-	session, renewalErrChan, err := rm.Store.NewSession(sessionName(id), nil)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Unable to get consul session. Halting \n%v\n", err)
-		os.Exit(1)
-	}
-
-	go func(errChan chan error) {
-		select {
-		case <-errChan:
-			fmt.Fprintf(os.Stderr, "Got renewal error, proceeding without a lock\n%v\n", err)
-		default:
-		}
-	}(renewalErrChan)
-
-	rcLock, err := rm.RCStore.LockForMutation(id, session)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Unable to lock RC for mutation. Halting.\n %v", err)
-		os.Exit(1)
-	}
-	defer rcLock.Unlock()
-
-	err = rm.RCStore.Disable(id)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Could not disable RC. %s \n %v", id, err)
-		os.Exit(1)
-	}
-
-	err = rm.deletePod()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Unable to delete pod. Please re-run this command to clean up. \n%v", err)
-		os.Exit(1)
-	}
-
-	err = rm.RCStore.AddDesiredReplicas(id, -1)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Unable to decrement RC count: %v", err)
-		os.Exit(1)
-	}
-
-	err = rm.RCStore.Enable(id)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Could not enable RC. %s \n %v", id, err)
-		os.Exit(1)
-	}
-
-	if err = session.Destroy(); err != nil {
-		fmt.Fprintf(os.Stderr, "Unable to destroy consul session. %v", err)
-	}
-
 	return nil
 }
 
-func (rm *P2RM) checkForManagingReplicationController() (bool, fields.ID, error) {
-	podLabels, err := rm.Labeler.GetLabels(labels.POD, rm.LabelID)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "unable to check node for labels: %v\n", err)
-		return false, "", err
+func handleUUIDPodRemoval(consulClient consulutil.ConsulClient, podUniqueKey types.PodUniqueKey, podID types.PodID, labeler labels.ApplicatorWithoutWatches) error {
+	if *nodeName != "" {
+		fmt.Fprintf(os.Stderr, "--node flag cannot be used with --pod-unique-key")
+		os.Exit(1)
 	}
 
-	if podLabels.Labels.Has(rc.RCIDLabel) {
-		return true, fields.ID(podLabels.Labels.Get(rc.RCIDLabel)), nil
-	}
+	rm := NewUUIDP2RM(consulClient, labeler)
 
-	return false, "", nil
+	// UUID pods are never managed by replication controllers
```

Comment:
This looks like a bug waiting to bite us next year sometime. Does it make sense to pull `checkForManagingReplicationController` up into some shared code?